docs/sources/next/using-k6/protocols/grpc.md

```diff
@@ -17,5 +17,322 @@ The binary format makes data transfer faster and more compact.
+const points = [
+  {
+    location: { latitude: 407838351, longitude: -746143763 },
+    name: 'Patriots Path, Mendham, NJ 07945, USA',
+  },
+  {
+    location: { latitude: 408122808, longitude: -743999179 },
+    name: '101 New Jersey 10, Whippany, NJ 07981, USA',
+  },
+  {
+    location: { latitude: 413628156, longitude: -749015468 },
+    name: 'U.S. 6, Shohola, PA 18458, USA',
+  },
+  {
+    location: { latitude: 419999544, longitude: -740371136 },
+    name: '5 Conners Road, Kingston, NY 12401, USA',
+  },
+  {
+    location: { latitude: 414008389, longitude: -743951297 },
+    name: 'Mid Hudson Psychiatric Center, New Hampton, NY 10958, USA',
+  },
+];
+
+export default () => {
+  if (__ITER == 0) {
+    client.connect('127.0.0.1:10000', { plaintext: true, reflect: true });
+  }
+
+  const stream = new Stream(client, 'main.RouteGuide/RecordRoute');
+
+  stream.on('data', (stats) => {
+    console.log(`Finished trip with ${stats.pointCount} points`);
+    console.log(`Passed ${stats.featureCount} features`);
+    console.log(`Travelled ${stats.distance} meters`);
+    console.log(`It took ${stats.elapsedTime} seconds`);
+  });
+
+  stream.on('end', () => {
+    client.close();
+    console.log('All done');
+  });
+
+  // send 3 random points
+  for (let i = 0; i < 3; i++) {
+    const point = points[Math.floor(Math.random() * points.length)];
+    pointSender(stream, point);
+  }
+
+  // close the client stream
+  stream.end();
+};
+
+const pointSender = (stream, point) => {
+  console.log(
+    `Visiting point ${point.name} ${point.location.latitude / COORD_FACTOR}, ${
+      point.location.longitude / COORD_FACTOR
+    }`
+  );
+
+  // send the location to the server
+  stream.write(point.location);
+
+  sleep(0.5);
+};
+```
+
+{{< /code >}}
+
+It establishes a connection to a gRPC server, creates a stream, and sends three random points. The server responds with statistics about the trip, which are logged to the console. The code also handles the end of the stream, closing the client and logging a completion message.
+
+### Bidirectional gRPC streaming
+
+In bi-directional streaming mode, both the client and the server may send multiple messages.
+
+From the API perspective, it combines the client and server streaming modes, so the code is similar to the examples above.
+
+### Streaming error handling
+
+To catch errors that occur during streaming, you can use the `error` event handler.
+
+The handler receives [an error object](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-net-grpc/stream/stream-error/).
+
+{{< code >}}
+
+```javascript
+import { Client, Stream } from 'k6/net/grpc';
+
+const client = new Client();
+const stream = new Stream(client, 'main.RouteGuide/RecordRoute');
+
+stream.on('error', function (e) {
+  // An error has occurred and the stream has been closed.
+  console.log('Error: ' + JSON.stringify(e));
+});
+```
+
+{{< /code >}}
+
+### Handling special cases in message Marshaling/Unmarshaling
```

Comment:
I'm wondering if that section deserves a dedicated page in `k6/net/grpc` JavaScript's API thinking:  

The main reason why I haven't put it there is that Marshaling/Unmarshaling is not an API. But probably, it is worth cross-linking from the Invoke/Stream.on pages to this section once it is merged (don't want to do this in a single PR since there is a #1472 PR which could cause a lot of conflicts)

WDYT?

Reply:
> But probably, it is worth cross-linking from the Invoke/Stream.on pages to this section once it is merged (don't want to do this in a single PR since there is a #1472 PR which could cause a lot of conflicts)

Yeah, let's do this :+1: